### PR TITLE
Improve DirectoryInfo.GetDirectories exception details

### LINKED ISSUE #8544: DirectoryInfo.GetDirectories does not list IOException
If, on a FAT32 partition, a corrupt file or directory entry exists, an `IOException` is thrown when calling `DirectoryInfo.GetDirectories()`:

```
System.IO.IOException: The file or directory is corrupted and unreadable. : 'D:\fakepath'
   at System.IO.Enumeration.FileSystemEnumerator`1.FindNextEntry()
   at System.IO.Enumeration.FileSystemEnumerator`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.IO.DirectoryInfo.GetDirectories(String searchPattern, EnumerationOptions enumerationOptions)
   at System.IO.DirectoryInfo.GetDirectories()
   at <>.FileBrowserPageViewModel.GetAllFoldersAndFiles() in <>\ViewModels\FileBrowserPageViewModel.cs:line 695
```

This is not listed in the documentation:

`DirectoryInfo.GetDirectories()` documentation as of Oct 21, 2022

### PR DESCRIPTION
**Pull request by adamsitnik**
fixes #8544

### CODE REVIEW

File: xml/System.IO/Directory.xml [hunk 1]
```diff
@@ -2405,7 +2405,7 @@ An I/O error occurred.</exception>
           <paramref name="path" /> is <see langword="null" />.</exception>
         <exception cref="T:System.IO.PathTooLongException">The specified path, file name, or both exceed the system-defined maximum length.</exception>
         <exception cref="T:System.IO.IOException">
-          <paramref name="path" /> is a file name.</exception>
+          <paramref name="path" /> is a file name or when file or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>
```
Comment (Jozkee) [inline xml/System.IO/Directory.xml:2408]: The "path is a file name" exception is unrelated to the new exception, that's why it should use the `-or-`. 
```xml
        <exception cref="T:System.IO.IOException"><paramref name="path" /> is a file name.
          
 -or- 
 
File or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>
```

Comment (adamsitnik) [inline xml/System.IO/Directory.xml:2408]:
```suggestion
          <paramref name="path" /> is a file name.

 -or- 
 
 File or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>    
```

File: xml/System.IO/Directory.xml [hunk 2]
```diff
@@ -2505,7 +2505,7 @@ An I/O error occurred.</exception>
           <paramref name="path" /> or <paramref name="searchPattern" /> is <see langword="null" />.</exception>
         <exception cref="T:System.IO.PathTooLongException">The specified path, file name, or both exceed the system-defined maximum length.</exception>
         <exception cref="T:System.IO.IOException">
-          <paramref name="path" /> is a file name.</exception>
+          <paramref name="path" /> is a file name or when file or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>
```
Comment (adamsitnik) [inline xml/System.IO/Directory.xml:2508]:
```suggestion
          <paramref name="path" /> is a file name.

 -or- 
 
 File or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>   
```

File: xml/System.IO/Directory.xml [hunk 3]
```diff
@@ -2589,7 +2589,7 @@ An I/O error occurred.</exception>
...
```
Comment (adamsitnik) [inline xml/System.IO/Directory.xml:2592]:
```suggestion
          <paramref name="path" /> is a file name.

 -or- 
 
 File or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>   
```

File: xml/System.IO/Directory.xml [hunk 4]
```diff
@@ -2685,7 +2685,7 @@ An I/O error occurred.</exception>
         <exception cref="T:System.UnauthorizedAccessException">The caller does not have the required permission.</exception>
         <exception cref="T:System.IO.PathTooLongException">The specified path, file name, or both exceed the system-defined maximum length.</exception>
         <exception cref="T:System.IO.IOException">
-          <paramref name="path" /> is a file name.</exception>
+          <paramref name="path" /> is a file name or when file or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>
```
Comment (adamsitnik) [inline xml/System.IO/Directory.xml:2688]:
```suggestion
          <paramref name="path" /> is a file name.

 -or- 
 
 File or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>   
```

**Review by Jozkee — APPROVED** [review]

**Pull request merged**